barcodecop/barcodecop.py

```diff
@@ -217,31 +219,29 @@ def main(arguments=None):
     # use bc1 to determine most common barcode
     bc1, bc2 = tee(bcseqs, 2)
 
-    # determine the most common barcode
-    barcode_counts = Counter(
-        [str(seq.seq) for seq in islice(bc1, args.snifflimit)])
+    iseqs = [str(seq.seq) for seq in islice(bc1, args.snifflimit)]
 
-    if not barcode_counts.most_common():
-        log.error('no reads, exiting')
-        if args.allow_empty:
-            return None
-        else:
-            raise SystemExit(1)
+    # determine the most common barcode
+    if iseqs:
+        barcode_counts = Counter(iseqs)
+    else:
+        barcode_counts = Counter({None: 0})
 
     barcodes, counts = list(zip(*barcode_counts.most_common()))
 
     most_common_bc = barcodes[0]
-    most_common_pct = 100 * float(counts[0]) / sum(counts)
+    most_common_pct = 100 * float(counts[0]) / (sum(counts) or 1)
```

Comment:
Might be better to just catch the error here

Reply:
https://github.com/nhoffman/barcodecop/pull/11/commits/bc77f3f996e4455d7390d01aeb7fbd68b23ccd31